update-doi-request/src/test/java/no/unit/nva/doi/requests/handlers/ApiUpdateDoiRequestHandlerTest.java

```diff
@@ -0,0 +1,341 @@
+        updateDoiRequest.setPublicationId(notExistingPublicationIdentifier);
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+        GatewayResponse<Problem> response =  sendRequest(updateDoiRequest, USERNAME_NOT_IMPORTANT);
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(HttpStatus.SC_NOT_FOUND)));
+
+        assertThat(details.getDetail(),
+            containsString(String.format(PUBLICATION_ID_NOT_FOUND_ERROR_FORMAT, notExistingPublicationIdentifier)));
+    }
+
+    @Test
+    public void handleRequestReturnsBadRequestWhenPublicationDoesNotHaveDoiRequest() throws IOException {
+        var publication = insertPublicationWithoutDoiRequest(
+            Clock.fixed(mockOneHourBefore, ZoneId.systemDefault()));
+
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        updateDoiRequest.setPublicationId(publication.getIdentifier().toString());
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+        GatewayResponse<Problem> response =  sendRequest(updateDoiRequest, publication.getOwner());
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(HttpStatus.SC_BAD_REQUEST)));
+
+        assertThat(details.getDetail(),
+            containsString("You must initiate creation of a DoiRequest before you can update it."));
+    }
+
+    @Test
+    public void handleRequestReturnsForbiddenExceptionWhenInputUsernameIsNotThePublicationOwner()
+            throws IOException {
+        final TestAppender appender = LogUtils.getTestingAppender(DynamoDBDoiRequestsService.class);
+        Publication publication = insertPublicationWithoutDoiRequest(
+            Clock.fixed(mockOneHourBefore, ZoneId.systemDefault()));
+
+        var updateDoiRequest = new ApiUpdateDoiRequest();
+        updateDoiRequest.setPublicationId(publication.getIdentifier().toString());
+        updateDoiRequest.setDoiRequestStatus(DoiRequestStatus.APPROVED);
+
+        GatewayResponse<Problem> response = sendRequest(updateDoiRequest, INVALID_USERNAME);
+
+        final Problem details = response.getBodyObject(Problem.class);
+
+        assertThat(response.getStatusCode(), Is.is(IsEqual.equalTo(HttpStatus.SC_FORBIDDEN)));
+
+        assertThatProblemDetailsDoesNotRevealSensitiveInformation(details, INVALID_USERNAME,
+                validUsername(publication));
+
+        assertThatLogsContainReasonForForbiddenMessage(appender, publication);
+    }
+
+    @Test
+    public void handleRequestReturnsForbiddenExceptionWhenUsernameIsNotInRequestContext()
+        throws IOException {
+        final TestAppender appender = LogUtils.getTestingAppender(UpdateDoiRequestHandler.class);
+        Publication publication = insertPublicationWithoutDoiRequest(
+            Clock.fixed(mockOneHourBefore, ZoneId.systemDefault()));
```

Comment:
Refactor this to `mockClock(Instant time)` or sth. This direction is used in many places in the code